### ISSUE TITLE
Error from isort while running pre-commit hooks

### ISSUE THREAD

**Issue by emilepetrone** [pydanny/cookiecutter-django#2502]
## What happened?

I set up a new project without Docker. When I tried to run the first commit, I am getting an isort issue on users that shouldn't be there.

```
Trim Trailing Whitespace.................................................Passed
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

<app>/users/urls.py:3:1: I001 isort found an import in the wrong position
<app>/users/urls.py:4:1: I001 isort found an import in the wrong position
<app>/users/urls.py:5:1: I001 isort found an import in the wrong position
<app>/users/urls.py:6:1: I001 isort found an import in the wrong position
<app>/users/urls.py:7:1: I001 isort found an import in the wrong position
```

I have not edited the `users/urls.py` file:

```
from django.urls import path

from .views import (
    user_detail_view,
    user_redirect_view,
    user_update_view,
)

app_name = "users"
urlpatterns = [
    path("~redirect/", view=user_redirect_view, name="redirect"),
    path("~update/", view=user_update_view, name="update"),
    path("<str:username>/", view=user_detail_view, name="detail"),
]
```

My `users/views.py` is also unchanged:

```
from django.contrib import messages
from django.contrib.auth import get_user_model
from django.contrib.auth.mixins import LoginRequiredMixin
from django.urls import reverse
from django.utils.translation import ugettext_lazy as _
from django.views.generic import DetailView, RedirectView, UpdateView

User = get_user_model()


class UserDetailView(LoginRequiredMixin, DetailView):

    model = User
    slug_field = "username"
    slug_url_kwarg = "username"


user_detail_view = UserDetailView.as_view()


class UserUpdateView(LoginRequiredMixin, UpdateView):

    model = User
    fields = ["name"]

    def get_success_url(self):
        return reverse("users:detail", kwargs={"username": self.request.user.username})

    def get_object(self):
        return User.objects.get(username=self.request.user.username)

    def form_valid(self, form):
        messages.add_message(
            self.request, messages.INFO, _("Infos successfully updated")
        )
        return super().form_valid(form)


user_update_view = UserUpdateView.as_view()


class UserRedirectView(LoginRequiredMixin, RedirectView):

    permanent = False

    def get_redirect_url(self):
        return reverse("users:detail", kwargs={"username": self.request.user.username})


user_redirect_view = UserRedirectView.as_view()
```

## What should've happened instead?

The first commit should not have an error with isort.

## Steps to reproduce

[//]: # (Any or all of the following:)
Follow the [non-docker instructions](https://cookiecutterdjango.readthedocs.io/en/latest/developing-locally.html).

[//]: # (* Host system configuration: OS, Docker & friends' versions etc.) 
Mac 10.15.3, Python 3.7.5

[//]: # (* Replay file https://cookiecutter.readthedocs.io/en/latest/advanced/replay.html)

```
{
  "cookiecutter": {
    "project_name": "SupLoca",
    "project_slug": "suploca",
    "description": "Support Local Businesses",
    "author_name": "Emile Petrone",
    "domain_name": "suploca.com",
    "email": "<my email>",
    "version": "0.1.0",
    "open_source_license": "Not open source",
    "timezone": "America/New_York",
    "windows": "n",
    "use_pycharm": "n",
    "use_docker": "n",
    "postgresql_version": "11.3",
    "js_task_runner": "None",
    "cloud_provider": "AWS",
    "mail_service": "Mailgun",
    "use_drf": "n",
    "custom_bootstrap_compilation": "n",
    "use_compressor": "n",
    "use_celery": "y",
    "use_mailhog": "y",
    "use_sentry": "y",
    "use_whitenoise": "n",
    "use_heroku": "n",
    "ci_tool": "None",
    "keep_local_envs_in_vcs": "y",
    "debug": "n",
    "_template": "https://github.com/pydanny/cookiecutter-django"
  }
}
```

[//]: # (* Logs)

```
Trim Trailing Whitespace.................................................Passed
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

<app>/users/urls.py:3:1: I001 isort found an import in the wrong position
<app>/users/urls.py:4:1: I001 isort found an import in the wrong position
<app>/users/urls.py:5:1: I001 isort found an import in the wrong position
<app>/users/urls.py:6:1: I001 isort found an import in the wrong position
<app>/users/urls.py:7:1: I001 isort found an import in the wrong position
```

**Comment by browniebroke** [comment]
This is coming from `flake8-isort`, which checks for isort while running flake8. Our isort config in located in the `.editorconfig` file:

https://github.com/pydanny/cookiecutter-django/blob/b5f2e161f69a8a417f2eebf6b1c0d2450dcde099/%7B%7Bcookiecutter.project_slug%7D%7D/.editorconfig#L15-L25

However, when running flake8 in pre-commit hook, we explicitly specify the config to be `setup.cfg`:

https://github.com/pydanny/cookiecutter-django/blob/b5f2e161f69a8a417f2eebf6b1c0d2450dcde099/%7B%7Bcookiecutter.project_slug%7D%7D/.pre-commit-config.yaml#L19

Maybe this is causing isort to use the `setup.cfg` instead of `.editorconfig`? Does it work if you remove the `args` line from `.pre-commit-config.yaml`?

**Comment by emilepetrone** [comment]
@browniebroke thanks for the comment. 

Removing `args` from `.pre-commit-config.yaml` doesn't solve it.  Here is what I ran against:

```
exclude: 'docs|node_modules|migrations|.git|.tox'
default_stages: [commit]
fail_fast: true

repos:
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: master
    hooks:
      - id: trailing-whitespace
        files: (^|/).+\.(py|html|sh|css|js)$

-   repo: local
    hooks:
      - id: flake8
        name: flake8
        entry: flake8
        language: python
        types: [python]
```

**Comment by emilepetrone** [comment]
If I remove 

```
-   repo: local
    hooks:
      - id: flake8
        name: flake8
        entry: flake8
        language: python
        types: [python]
        args: ['--config=setup.cfg']
``` 
It will allow the commit.

**Comment by browniebroke** [comment]
Ok, so our pre-commit hook config is a bit messy. We use the `repo: local` which means it's using whatever is installed in your system under `flake8`. So depending on your system, it will or will not work. 

While running it, I had other errors coming from `flake8-docstring`, which I have installed somewhere.

Instead, the pre-commit config should use a config much closer to the recommended one, to install from a git repo, with a pinned version & extra dependencies. This seems to work on my machine:

```
- repo: https://gitlab.com/pycqa/flake8
  rev: 3.7.9
  hooks:
  - id: flake8
    additional_dependencies: [flake8-isort==2.9.0]
```

Could you please try it out?

**Comment by emilepetrone** [comment]
@browniebroke 

That worked for me. Specifically here is what I ran:

`exclude: 'docs|node_modules|migrations|.git|.tox'
default_stages: [commit]
fail_fast: true

repos:
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: master
    hooks:
      - id: trailing-whitespace
        files: (^|/).+\.(py|html|sh|css|js)$

-   repo: https://gitlab.com/pycqa/flake8
    rev: 3.7.9
    hooks:
      - id: flake8
        name: flake8
        entry: flake8
        language: python
        types: [python]
        args: ['--config=setup.cfg']
        additional_dependencies: [flake8-isort==2.9.0]
`

And the result was:

`
[INFO] Initializing environment for https://gitlab.com/pycqa/flake8.
[INFO] Initializing environment for https://gitlab.com/pycqa/flake8:flake8-isort==2.9.0.
[INFO] Installing environment for https://gitlab.com/pycqa/flake8.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Trim Trailing Whitespace.............................(no files to check)Skipped
flake8...............................................(no files to check)Skipped
[master eb1b6de] Pre commit changes
 1 file changed, 3 insertions(+), 2 deletions(-)
`

**Comment by demestav** [comment]
> Instead, the pre-commit config should use a config much closer to the recommended one, to install from a git repo, with a pinned version & extra dependencies.

@browniebroke shouldn't we use the flake8 setup that we already define in the *requirements/local.txt*?

**Comment by browniebroke** [comment]
Yes, ideally we want to reuse the same version and extensions... I don't know if there is a good way to point at a specific flake8 installation reliabely, but the way we do it currently is definitely not reliable (see my previous explanation).

**Comment by browniebroke** [comment]
I've made a pull request #2584 which should fix the issues, simplify the config & add Black. It's not perfect as we are duplicating the version from `requirements/local.txt`, but it should improve the current situation, I think.

**Comment by alexche77** [comment]
I am not quite sure on what to do, but I will leave this here:

```
git commit -am "initial commit"
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Check Yaml...............................................................Passed
black....................................................................Passed
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

config/settings/production.py:2:1: I003 isort expected 1 blank line in imports, found 0
config/settings/production.py:5:1: I001 isort found an import in the wrong position
loan_tracker/users/tests/test_views.py:8:1: I001 isort found an import in the wrong position
loan_tracker/users/tests/test_views.py:9:1: I001 isort found an import in the wrong position
loan_tracker/users/tests/test_views.py:10:1: I001 isort found an import in the wrong position
loan_tracker/users/tests/test_views.py:11:1: I001 isort found an import in the wrong position
loan_tracker/users/tests/test_views.py:12:1: I001 isort found an import in the wrong position

```

Pre-commit hook is failing

**Comment by zwass** [comment]
I get the error when running

```
docker-compose -f local.yml run --rm django flake8
```

**Comment by Shannanigans** [comment]
Not well versed in these issues, but I have been digging around trying to figure this out and I believe the issue is related to the fix in the latest flake8-isort==3.0.1 release.

https://pypi.org/project/flake8-isort/
